### ISSUE TITLE
False positives with custom properties

### ISSUE THREAD

**Issue by draeklae** [PyCQA/pylint#1601]
### Example code
```
class custom_prop(property):
    '''empty'''

class MyClass(object):
    @custom_prop
    def my_prop(self):
        return [1, 2, 3]

    def my_funct(self):
        for i in self.my_prop:
            print(i)
        self.my_prop.append(42)
        return self.my_prop[0]
```

### Current behavior
This gives "not-an-iterable", "no-member" and "unsubscriptable-object" errors, which don't show up if I replace the "custom_prop" decorator with the "property" bultin.

The errors also don't appear if I convert MyClass into an ABC and use "abc.abstractproperty" as a decorator.

### Expected behavior
No errors.

### pylint --version output
pylint 1.8.0, 
astroid 1.5.3
Python 3.6.2 (default, Jul 20 2017, 03:52:27) 
[GCC 7.1.1 20170630]

**Comment by PCManticore** [comment]
Fixed in astroid's master. Thanks for the report!
